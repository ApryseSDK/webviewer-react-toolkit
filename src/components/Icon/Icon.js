'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Icon = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var icons = tslib_1.__importStar(require('./../../icons'));
exports.Icon = (0, react_1.forwardRef)(function(_a, ref) {
  var icon = _a.icon,
    svgProps = _a.svgProps,
    className = _a.className,
    children = _a.children,
    props = tslib_1.__rest(_a, ['icon', 'svgProps', 'className', 'children']);
  var iconClass = (0, classnames_1.default)('ui__base ui__icon', className);
  var child = (0, react_1.useMemo)(
    function() {
      if (children !== undefined) return children;
      if (icons === undefined) return undefined;
      var IconChild = icons[icon];
      return react_1.default.createElement(
        IconChild,
        tslib_1.__assign({}, svgProps),
      );
    },
    [children, icon, svgProps],
  );
  return react_1.default.createElement(
    'i',
    tslib_1.__assign({}, props, { className: iconClass, ref: ref }),
    child,
  );
});
//# sourceMappingURL=Icon.js.map
