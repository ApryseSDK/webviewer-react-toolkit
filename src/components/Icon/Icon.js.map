{"version":3,"file":"Icon.js","sourceRoot":"","sources":["Icon.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAwF;AACxF,2DAAuC;AAsB1B,QAAA,IAAI,GAAG,IAAA,kBAAU,EAC5B,UAAC,EAAiD,EAAE,GAAG;IAApD,IAAA,IAAI,UAAA,EAAE,QAAQ,cAAA,EAAE,SAAS,eAAA,EAAE,QAAQ,cAAA,EAAK,KAAK,sBAA/C,6CAAiD,CAAF;IAC9C,IAAM,SAAS,GAAG,IAAA,oBAAU,EAAC,mBAAmB,EAAE,SAAS,CAAC,CAAC;IAE7D,IAAM,KAAK,GAAG,IAAA,eAAO,EAAC;QACpB,IAAI,QAAQ,KAAK,SAAS;YAAE,OAAO,QAAQ,CAAC;QAC5C,IAAI,KAAK,KAAK,SAAS;YAAE,OAAO,SAAS,CAAC;QAC1C,IAAM,SAAS,GAAG,KAAK,CAAC,IAAK,CAAC,CAAC;QAC/B,OAAO,8BAAC,SAAS,uBAAK,QAAQ,EAAI,CAAC;IACrC,CAAC,EAAE,CAAC,QAAQ,EAAE,IAAI,EAAE,QAAQ,CAAC,CAAC,CAAC;IAE/B,OAAO,CACL,wDAAO,KAAK,IAAE,SAAS,EAAE,SAAS,EAAE,GAAG,EAAE,GAAG,KACzC,KAAK,CACJ,CACL,CAAC;AACJ,CAAC,CACF,CAAC"}