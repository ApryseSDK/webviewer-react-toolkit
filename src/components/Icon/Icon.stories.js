'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var react_1 = tslib_1.__importDefault(require('react'));
var selectIcon_1 = require('../../storybook-helpers/knobs/selectIcon');
var Icon_1 = require('../Icon');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/Icon',
  component: Icon_1.Icon,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(Icon_1.Icon, {
    icon: (0, selectIcon_1.selectIcon)('icon', 'Close'),
  });
};
exports.Basic = Basic;
//# sourceMappingURL=Icon.stories.js.map
