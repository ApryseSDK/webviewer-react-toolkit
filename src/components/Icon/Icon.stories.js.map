{"version":3,"file":"Icon.stories.js","sourceRoot":"","sources":["Icon.stories.tsx"],"names":[],"mappings":";;;;AAAA,wDAA0B;AAC1B,uEAAsE;AACtE,gCAA+B;AAC/B,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,iBAAiB,EAAE,SAAS,EAAE,WAAI,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE9E,IAAM,KAAK,GAAG,cAAM,OAAA,8BAAC,WAAI,IAAC,IAAI,EAAE,IAAA,uBAAU,EAAC,MAAM,EAAE,OAAO,CAAC,GAAI,EAA3C,CAA2C,CAAC;AAA1D,QAAA,KAAK,SAAqD"}