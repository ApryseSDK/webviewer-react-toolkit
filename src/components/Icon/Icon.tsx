import classnames from 'classnames';
import React, { forwardRef, HTMLAttributes, ReactNode, SVGProps, useMemo } from 'react';
import * as icons from './../../icons';

export type AvailableIcons = keyof typeof icons;

export interface IconProps extends HTMLAttributes<HTMLElement> {
  /**
   * Specify one of the included icons from the toolkit. If provided, do not add
   * `children` or they will override this.
   */
  icon?: AvailableIcons;
  /**
   * Props that will be passed to the included icons. This will not be used if
   * `children` is provided.
   */
  svgProps?: SVGProps<SVGSVGElement>;
  /**
   * Provide a custom child instead of a provided icon. Will override `icon` if
   * provided.
   */
  children?: ReactNode;
}

export const Icon = forwardRef<HTMLButtonElement, IconProps>(
  ({ icon, svgProps, className, children, ...props }, ref) => {
    const iconClass = classnames('ui__base ui__icon', className);

    const child = useMemo(() => {
      if (children !== undefined) return children;
      if (icons === undefined) return undefined;
      const IconChild = icons[icon!];
      return <IconChild {...svgProps} />;
    }, [children, icon, svgProps]);

    return (
      <i {...props} className={iconClass} ref={ref}>
        {child}
      </i>
    );
  },
);
