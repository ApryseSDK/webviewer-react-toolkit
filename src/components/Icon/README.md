Icon allows you to specify one of the included icons using the `icon` prop, or
add your own custom child icon.
