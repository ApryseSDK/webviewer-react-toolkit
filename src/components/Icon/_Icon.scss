.ui__icon {
  @include flex-center;
}
