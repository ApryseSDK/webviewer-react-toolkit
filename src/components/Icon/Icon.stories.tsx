import React from 'react';
import { selectIcon } from '../../storybook-helpers/knobs/selectIcon';
import { Icon } from '../Icon';
import readme from './README.md';

export default { title: 'Components/Icon', component: Icon, parameters: { readme } };

export const Basic = () => <Icon icon={selectIcon('icon', 'Close')} />;
