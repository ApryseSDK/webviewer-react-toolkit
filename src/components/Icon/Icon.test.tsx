import { shallow } from 'enzyme';
import React from 'react';
import { Icon } from '../Icon';

describe('Icon component', () => {
  it('renders its contents', () => {
    const icon = shallow(<Icon icon="Close" />);
    expect(icon.find('.ui__icon')).toHaveLength(1);
  });

  it('snapshot renders default icon', () => {
    const icon = shallow(<Icon icon="Close" />);
    expect(icon).toMatchSnapshot();
  });
});
