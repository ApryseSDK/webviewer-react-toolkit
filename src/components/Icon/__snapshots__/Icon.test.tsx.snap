// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component snapshot renders default icon 1`] = `
<i
  className="ui__base ui__icon"
>
  <Close />
</i>
`;
